app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -19,28 +19,32 @@ def initialize(hosts, template_invocation = nil)
     # more advanced cases). Therefore we have only the 'all' group
     # with all hosts.
     def to_hash
+      to_hash_with_secrets_redacted(false)
+    end
+
+    def to_hash_with_secrets_redacted(redact_secrets = true)
```

Comment:
Why do we always hide the value here? What if the user has permission to see the hidden values?

Reply:
Which permission would this be? I don't think such a permission exists.

Reply:
admin user/any user with `edit_ansible_variables` permission.

Reply:
Ah. Thanks. I thought there is a permission whether to see hidden values or not

Reply:
I am hiding the value by default because like this, a conscious decision to show the values is required by the user. 
That's also how it works in **Configure**>**Ansible**>**Variables**>**<variable>**.

Reply:
Correct. However, in our case, we are changing existing behavior. For example, if a user is using the API to read the hidden variable values, with our new changes, they won't be able to do that anymore, even with the right permissions.

Reply:
You are right. The hidden values were automatically hidden in the inventory by default. This is now fixed. 
As before, Inventories list hidden values by default. One may opt to include `redact_secrets=true` in the request to redact the secret values.
This is still done in the UI, as this, IMHO, is in line with the idea behind hidden values. Either way, a user needs the `view_hosts` permission to view a host's inventory, redacted or not.